### ISSUE TITLE
how to do Relation Extraction  after using the components(BERN2:http://bern2.korea.ac.kr/) of this prediction entity.

### ISSUE THREAD

**Issue by zhangweizhenGitHub** [dmis-lab/biobert#179]
hello,[jhyuklee](https://github.com/dmis-lab/biobert/commits?author=jhyuklee)
       how to do Relation Extraction  after using the components(BERN2:http://bern2.korea.ac.kr/) of this prediction entity.
       Does this project support this?
       and i think RE :  input plain text with NER,output is Relationship ,Is that right?

**Comment by wonjininfo** [comment]
Hi @zhangweizhenGitHub , 
We do not have a plain-text to RE system in our lab (like BERN in the NER task). 

You may be interested in this: I have recently worked on the RE task using LMs and HuggingFace and PyTorch and made our code available on https://github.com/dmis-lab/BioRE-drugprot-kuaz 
However, I did not make preprocessing code available yet. (I wish I can do it soon but has a list of things to do for my graduation these days - Apologies for this)
So you will need to write some code to pre-process your input data.
 
In short, to predict relation classes for a plain text, you need to 
1. run BERN2 to recognize named entities
2. pre-process your input data (post-NER) to match the format (check `pre-processed datasets` in the BioRE repo)
3. use a trained model to predict relation classes for your input data

Thank you for your interest in our work!
Best, 
Wonjin

**Comment by zhangweizhenGitHub** [comment]
Got it. very thanks for your replay. It's admirable that you have to maintain this project in addition to your graduation。
In short, to predict relation classes for a plain text:
     1. In short, to predict relation classes for a plain text(Done)
     2. I need to find some labeded data about my focus on domain,such as this repo(Drugprot),and maybe other domain(About a specific disease target)，I think It is better to use that domain's data to train model.right?(The challenge here is to label the text data in these specific fields)
     3. according to pre-processed datasets fomat -->use a trained model to predict new plain text relation 
Above is I want to confirm my idea with you 
best, 
Frank

**Comment by wonjininfo** [comment]
@zhangweizhenGitHub  Yes, your idea is correct. 
And about 2, I think finding the training data exactly coherent with your interest will be the best.
